Sources/Knit/Module/Container+AbstractRegistration.swift

```diff
@@ -171,12 +183,23 @@ extension Container {
             toService entry: ServiceEntry<Service>,
             withName name: String?
         ) {
-            let id = RegistrationKey(typeIdentifier: ObjectIdentifier(Type.self), name: name)
+            let id = RegistrationKey(
+                typeIdentifier: ObjectIdentifier(Type.self),
+                name: name,
+                concurrency: .unknown
```

Comment:
If we were to internalize and modify Swinject we could be forwarding the concurrency information we have on our concrete registrations through to this behavior